### PR TITLE
Scout cloak signals

### PR DESCRIPTION
**Pull request by spookydonut**
## About The Pull Request
converts scout cloak to use signals

## Changelog
:cl:
fix: fixed a bug that allowed a cloak scout to use a melee weapon without shimmering
/:cl:

### CODE REVIEW

**Comment by DominikPanic** [comment]
Have you tested this locally?

File: code/game/objects/items/storage/backpack.dm [hunk 1]
```diff
@@ -27,12 +27,11 @@
...
 
-/obj/item/storage/backpack/attackby(obj/item/I, mob/user, params)
+/obj/item/storage/backpack/attackby(obj/item/W, mob/user)
 	. = ..()
 	
-	if(use_sound)
-		playsound(loc, use_sound, 15, 1, 6)
-
+	if (use_sound)
+		playsound(src.loc, src.use_sound, 15, 1, 6)
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]:
```suggestion
		playsound(loc, use_sound, 15, 1, 6)
```

File: code/game/objects/items/storage/backpack.dm [hunk 2]
```diff
@@ -27,12 +27,11 @@
 			return
 	..()
 
-/obj/item/storage/backpack/attackby(obj/item/I, mob/user, params)
+/obj/item/storage/backpack/attackby(obj/item/W, mob/user)
```
Comment (Rohesie) [inline code/game/objects/items/storage/backpack.dm]:
```suggestion
/obj/item/storage/backpack/attackby(obj/item/I, mob/user, params)
```

File: code/game/objects/items/storage/backpack.dm (hunk 1 above)
Comment (spookydonut) [reply code/game/objects/items/storage/backpack.dm]: oops

File: code/game/objects/items/storage/backpack.dm (hunk 2 above)
Comment (spookydonut) [reply code/game/objects/items/storage/backpack.dm]: ooops

**Comment by Rohesie** [comment]
```
[21:18:00] Runtime in unsorted.dm, line 37: human_attachment_fired overridden. Use override = TRUE to suppress this warning
proc name: stack trace (/datum/proc/stack_trace)
usr: CKEY/(Tasmin Elliot)
usr.loc: (Hallway Port (114, 50, 3))
src: the M68 Thermal Cloak (/obj/item/storage/backpack/marine/satchel/scout_cloak/scout)
src.loc: Tasmin Elliot (/mob/living/carbon/human)
call stack:
the M68 Thermal Cloak (/obj/item/storage/backpack/marine/satchel/scout_cloak/scout): stack trace("human_attachment_fired overrid...")
the M68 Thermal Cloak (/obj/item/storage/backpack/marine/satchel/scout_cloak/scout): RegisterSignal(Tasmin Elliot (/mob/living/carbon/human), /list (/list), /datum/callback (/datum/callback), 0)
the M68 Thermal Cloak (/obj/item/storage/backpack/marine/satchel/scout_cloak/scout): camouflage()
the M68 Thermal Cloak (/obj/item/storage/backpack/marine/satchel/scout_cloak/scout): Toggle M68 Thermal Camouflage()
```

**Pull request merged**